app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,64 @@
+class AlbumsController < ApplicationController
+  def get_album
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def index
+    @albums = Album.all
+    @albums_sort = @albums.sort_by{ |album| album[:rank] }.reverse
+  end
+
+  def show
+    get_album
+  end
+
+  def new
+    @album = Album.new
+    @title = "Add New album"
+    @action = "create"
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    @album.update(:rank => 0)
+    if @album.save
```

Comment:
You have already attempted to create and update the album by the time you get here. Although this code is functional, it's not a great practice to try saving something after you've already tried saving it twice. (Both `create` and `update` will trigger SQL interactions with the database. You can get a closer view of what's going on here by trying out these transactions in your rails console or watching the rails server output as you interact with the `albums#create` controller action.)

There are a couple things you could to here to DRY this up:

1. Instead of updating the album to have a rank of 0 after creating it, you could [set a default value](http://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/change_column_default) of 0 in your database.
2. Instead of catching errors on line 25 by trying to save again, you could [check for validity directly](http://guides.rubyonrails.org/active_record_validations.html#valid-questionmark-and-invalid-questionmark) with `@album.valid?`, which will return false if errors are present.